pkg/xds/generator/ingress_generator.go

```diff
@@ -70,13 +71,26 @@ func (i IngressGenerator) Generate(
 	if err != nil {
 		return nil, err
 	}
-	if len(listener.(*envoy_listener_v3.Listener).FilterChains) > 0 {
+
+	hasFilterChain := len(listener.(*envoy_listener_v3.Listener).FilterChains) > 0
+	if !hasFilterChain {
+		var blackHoleCluster envoy_common.NamedResource
+		listener, blackHoleCluster, err = zoneproxy.GenerateBlackHoleResources(proxy.APIVersion, address, port)
```

Comment:
Thinking about this in the context of: https://github.com/kumahq/kuma/issues/1907

Would it make sense to use: https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/direct_response_filter this way we don't add a cluster we don't need and can also implement healthchecks quite easily.

WDYT?

Reply:
What should we send into the response then?

Reply:
Hmm could be a json with zoneIngress name and zone. @jakubdyszkiewicz any idea?